src/accelerate/test_utils/testing.py

```diff
@@ -488,6 +491,10 @@ def tee(line, sink, pipe, label=""):
 
 
 def execute_subprocess_async(cmd, env=None, stdin=None, timeout=180, quiet=False, echo=True) -> _RunOutput:
+    # Cast everything in `cmd` to a string
+    for i, c in enumerate(cmd):
+        if isinstance(c, Path):
+            cmd[i] = str(c)
```

Comment:
It must always be mutable here, which I'm open to adding a type hint for but this is known and a repeated pattern. 

And while true, is a must to keep things clean like you would like, without a blanket convert across everything.

Reply:
I'm not quite sure why everything couldn't be cast though? (Also, I guess `run_command` should use the same pattern..?)

Reply:
What if a user decided to try passing in an `Accelerator` object instead? :)

Reply:
Well, I'd probably go with a full-on

```python
def _check_subprocess_args(cmd) -> list:
   for arg in cmd:
       if not isinstance(arg, (str, pathlib.Path)):
           raise TypeError(f"Subprocess argument must be a string or path, not {arg!r}")
       yield str(arg)
```
to get a clear early `TypeError` from the testing utility code instead of `subprocess.*` raising on a non-string argument later on :)